lib/gettext/extractor.ex

```diff
@@ -318,10 +318,15 @@ defmodule Gettext.Extractor do
     messages = old_and_merged ++ unique_new
 
     messages =
-      if gettext_config[:sort_by_msgid] do
-        Enum.sort_by(messages, & &1.msgid)
-      else
-        messages
+      cond do
+        gettext_config[:sort_by_msgid] ->
+          Enum.sort_by(messages, & &1.msgid)
+
+        gettext_config[:sort_by_msgid_case_insensitive] ->
+          Enum.sort_by(messages, fn m -> Enum.map(m.msgid, &String.downcase/1) end)
+
+        true ->
+          messages
       end
```

Comment:
To make sure that the line splitting of the message has no impact on the sorting, I would recommend changing this a bit:

```suggestion
      cond do
        gettext_config[:sort_by_msgid] ->
          Enum.sort_by(messages, &IO.chardata_to_string(&1.msgid))

        gettext_config[:sort_by_msgid_case_insensitive] ->
          Enum.sort_by(messages, &(&1 |> IO.chardata_to_string() |> String.downcase()))

        true ->
          messages
      end
```

Reply:
We were already sorting without `IO.chardata_to_string/1`. Do we know if it has ever caused issues? I think in general we should fix that in another PR and add test cases for it.

Reply:
@whatyouhide That’s something I should’ve probably done in #307 and spotted now when reading the changes.

I‘m not aware of issues so far but I can see how it coupd be an issue.

I can open a PR and add some tests if you prefer.

Reply:
Yeah, let's do that @maennchen, thanks 👍

Reply:
Done: https://github.com/elixir-gettext/gettext/pull/331

This PR will need a merge after #331 is merged.

Reply:
Merged #331 👍